core/dbt/task/deps.py

```diff
@@ -59,40 +121,166 @@ def track_package_install(
+
+        return packages_yml
+
+    def add(self):
+        packages_yml_filepath = (
+            f"{self.project.project_root}/{self.project.packages_specified_path}"
+        )
+
+        if not system.path_exists(packages_yml_filepath):
+            raise dbt.exceptions.DbtRuntimeError("No packages definition file.")
+
+        if not self.args.package_version and self.args.source != "local":
+            raise dbt.exceptions.DbtRuntimeError(
+                f"Version is required to add a package when source is {self.args.source}"
+            )
+
+        new_package_entry = _create_packages_yml_entry(
+            self.args.package, self.args.package_version, self.args.source
+        )
+
+        with open(packages_yml_filepath, "r") as user_yml_obj:
+            packages_yml = yaml.safe_load(user_yml_obj)
+            packages_yml = self.check_for_duplicate_packages(packages_yml)
+            packages_yml["packages"].append(new_package_entry)
+
+        self.project.packages.packages = package_config_from_data(packages_yml).packages
+
+        if packages_yml:
+            with open(packages_yml_filepath, "w") as pkg_obj:
+                pkg_obj.write(
+                    yaml.dump(packages_yml, Dumper=dbtPackageDumper, default_flow_style=False)
+                )
+
+                fire_event(
+                    DepsAddPackage(
+                        package_name=self.args.package,
+                        version=self.args.package_version,
+                        packages_filepath=packages_yml_filepath,
+                    )
+                )
+
+    def lock(self) -> None:
+        lock_filepath = f"{self.project.project_root}/{PACKAGE_LOCK_FILE_NAME}"
+
         packages = self.project.packages.packages
+        packages_installed: Dict[str, Any] = {"packages": []}
+
         if not packages:
             fire_event(DepsNoPackagesFound())
             return
 
         with downloads_directory():
-            final_deps = resolve_packages(packages, self.project, self.cli_vars)
+            resolved_deps = resolve_packages(packages, self.project, self.cli_vars)
+
+        # this loop is to create the package-lock.yml in the same format as original packages.yml
+        # package-lock.yml includes both the stated packages in packages.yml along with dependent packages
+        for package in resolved_deps:
+            lock_entry = _create_packages_yml_entry(
+                package.name, package.get_version(), package.source_type()
+            )
+            packages_installed["packages"].append(lock_entry)
+        packages_installed[PACKAGE_LOCK_HASH_KEY] = _create_sha1_hash(
+            self.project.packages.packages
+        )
+
+        with open(lock_filepath, "w") as lock_obj:
+            yaml.safe_dump(packages_installed, lock_obj)
+
+        fire_event(DepsLockUpdating(lock_filepath=lock_filepath))
+
+    def run(self) -> None:
+        if self.args.ADD:
```

Comment:
super nit: using varying conventions for accessing attributes on self.args in this method. The rest of the method uses lowercase, and I believe that's generally the case in the repo. so I'd use `self.args.add` here as well